wcfsetup/install/files/lib/acp/form/PackageUpdateServerAddForm.class.php

```diff
@@ -31,153 +41,176 @@ class PackageUpdateServerAddForm extends AbstractForm
     public $neededPermissions = ['admin.configuration.package.canEditServer'];
 
     /**
-     * server url
-     * @var string
-     */
-    public $serverURL = '';
-
-    /**
-     * server login username
-     * @var string
+     * @inheritDoc
      */
-    public $loginUsername = '';
+    public $objectActionClass = PackageUpdateServerAction::class;
 
     /**
-     * server login password
-     * @var string
+     * @inheritDoc
      */
-    public $loginPassword = '';
+    public $objectEditLinkController = PackageUpdateServerEditForm::class;
 
     /**
      * @inheritDoc
      */
-    public function readFormParameters()
+    protected function createForm()
     {
-        parent::readFormParameters();
-
-        if (isset($_POST['serverURL'])) {
-            $this->serverURL = StringUtil::trim($_POST['serverURL']);
-        }
-        if (isset($_POST['loginUsername'])) {
-            $this->loginUsername = $_POST['loginUsername'];
-        }
-        if (isset($_POST['loginPassword'])) {
-            $this->loginPassword = $_POST['loginPassword'];
+        parent::createForm();
+
+        $passwordPlaceholder = '';
+        if (
+            $this->formAction !== IFormDocument::FORM_MODE_CREATE
+            && $this->formObject instanceof PackageUpdateServer
+            && $this->formObject->loginPassword !== ''
+        ) {
+            $passwordPlaceholder = 'wcf.acp.updateServer.loginPassword.noChange';
         }
+
+        $this->form->appendChildren([
+            FormContainer::create('data')
+                ->appendChildren([
+                    UrlFormField::create('serverURL')
+                        ->label('wcf.acp.updateServer.serverURL')
+                        ->pattern('https://.+')
```

Comment:
```suggestion
```
Please remove the pattern. The error handling by the browser is “non-great”, making it totally non-obvious what the issue is (the lack of 's').